src/pages/RecipeDetails.js

```diff
@@ -9,13 +9,18 @@ import AddIcon from '@material-ui/icons/Add';
 import Fab from '@material-ui/core/Fab';
 import { useHistory } from 'react-router-dom';
 import { RecipeDataContext } from '../data/RecipeDataContext';
+import {IngredientsList} from '../components/IngredientsList'
 
 export default function RecipeDetails() {
+    
     const styles = useStyles();
     const { currentRecipe } = useContext(RecipeDataContext);
+    const [ingredients,setIngredients] = useState(currentRecipe.ingredients)
```

Comment:
The currentRecipe can be null, so you should add null check. Do the same thing in IngredientsList as the null still can be passed to that component.